src/api/user.js

```diff
@@ -36,6 +36,46 @@ module.exports = {
     })
   },
 
+  checkUserNameExist (username, callback, error) {
+    User.findOne({
+      'username': username
+    }).toArray(function (err, result) {
+      if (err) {
+        error(err)
+      } else {
+        callback(result.length)
```

Comment:
This should return a boolean value (e.g. `true`, `false`) but not an integer
```suggestion
        callback(result.length === 1)
```